bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BTypes.java

```diff
@@ -62,6 +63,11 @@
     public static BType typePureType = new BUnionType(Arrays.asList(typeAnydata, typeError));
     public static BType typeAllType = new BUnionType(Arrays.asList(typeAny, typeError));
     public static BType typeHandle = new BHandleType(TypeConstants.HANDLE_TNAME, new BPackage(null, null, null));
+    public static BRecordType stringItrNextReturnType = new BRecordType(TypeConstants.ITERATOR_NEXT_RETURN_TYPE,
+            new BPackage(null, null, null), 0, true, TypeFlags.asMask(TypeFlags.ANYDATA,
+            TypeFlags.PURETYPE));
+    public static BRecordType xmlItrNextReturnType = new BRecordType(TypeConstants.ITERATOR_NEXT_RETURN_TYPE,
+            new BPackage(null, null, null), 0, true, 0);
```

Comment:
Why are we setting 0 for type flags here? XML and string are both anydata and pure?

Reply:
fixed